webxrlayers-1.bs

```diff
@@ -769,21 +771,26 @@ is initialized.
 <pre class="idl">
 dictionary XRProjectionLayerInit {
   XRTextureType textureType = "texture";
+  GLenum colorFormat = 0x1907; // RGBA
```

Comment:
Any concerns that the default here may not align with the advertised preferred format? I was considering doing the same thing with WebGPU layers, but was a bit concerned about both requiring a specific format and choosing an arbitrary one that may not be optimal. That said, it does make layer init a lot easier for simply cases if there is a default here.

Reply:
Are there platforms where plain RGBA is not the preferred format?
Do you think we should have an attribute that returns the preferred format (and then default to that in the absence of the colorFormat entry in the init structure)?

Reply:
> Are there platforms where plain RGBA is not the preferred format?

I think BGRA is preferred on some devices? Regardless, I doubt it makes a huge difference performance difference for rendering? It's probably a bigger concern when doing texture copies.

> Do you think we should have an attribute that returns the preferred format

My intent with WebGPU layers was to follow OpenXR's convention of having the supported list be returned in order of preference.

> (and then default to that in the absence of the colorFormat entry in the init structure)

The idea of defaulting to the preferred format if the `colorFormat` is `null` is intriguing, but I worry that it would create confusion when paired with the `depthStencilFormat` (where we're discussing having `null` mean "Don't allocate a texture"). Also, in some cases where the developer is directly providing image data via `texSubImage2d()` or similar having the format shift out from under them may lead to situations where their code works on one device but shows "corrupted" image data on another that they didn't test on. In that regard a stable default, even if not the optimal format, is probably better.

So yeah, I think I just talked myself out of my concerns. And RGBA certainly seems like a logical default.